hikari/api/special_endpoints.py

```diff
@@ -1869,11 +1894,19 @@ def add_button(
         """
 
     @abc.abstractmethod
-    def add_select_menu(self: _T, custom_id: str, /) -> SelectMenuBuilder[_T]:
+    def add_select_menu(
+        self: _T,
+        type_: typing.Union[components_.ComponentType, int, str],
+        # These have default during the deprecation period for backwards compatibility, as custom_id
+        # used to come first
+        custom_id: str = "",
+    ) -> SelectMenuBuilder[_T]:
         """Add a select menu component to this action row builder.
 
         Parameters
         ----------
+        type_ : typing.Union[hikari.components.ComponentType, int]
+            The type for the select menu.
```

Comment:
I dont feel like its worth documenting it. It will raise a deprecation warning for those that decide to pass `custom_id` first, but wont complain at the new way

Reply:
Would it not be useful for the docs to actually describe what the deprecated behaviour is to help ppl understand what the deprecation warning is saying when they actually see it though